cupy/_environment.py

```diff
@@ -476,10 +476,10 @@ def _detect_duplicate_installation():
         'cupy-rocm-4-2',
         'cupy-rocm-4-3',
         'cupy-rocm-5-0',
-    ]
-    cupy_installed = [
-        name for name in known
-        if list(importlib.metadata.distributions(name=name))]
+    }
+    installed_names = {d.metadata["Name"]
+                                     for d in importlib.metadata.distributions()}
```

Comment:
autopep8 but account for weird github indenting
```suggestion
    installed_names = {d.metadata["Name"]
                       for d in importlib.metadata.distributions()}
```